pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/default.nix

```diff
@@ -0,0 +1,134 @@
+{ mkDerivation, lib, fetchFromGitHub, fetchsvn, fetchpatch
+, pkgconfig, pythonPackages, cmake, ninja, gcc9, qtbase
+, qtimageformats, gtk3, libappindicator-gtk3, libnotify
+, xdg_utils, ffmpeg_4, openalSoft, lzma, lz4, zlib, minizip
+, openssl, libtgvoip, rlottie-tdesktop, range-v3
+}:
+
+with lib;
+
+mkDerivation rec {
+  pname = "kotatogram-desktop";
+  version = "1.1.1";
+
+  src = fetchFromGitHub {
+    owner = "kotatogram";
+    repo = "kotatogram-desktop";
+    rev = "k${version}";
+    sha256 = "156r229dpxnyaxxgp1dr4vz8jla5p7r0dydhgdgk2yx1p11z6m3w";
+    fetchSubmodules = true;
+  };
+
+  # Arch patches (svn export telegram-desktop/trunk)
+  archPatches = fetchsvn {
+    url = "svn://svn.archlinux.org/community/telegram-desktop/trunk";
+    # svn log svn://svn.archlinux.org/community/telegram-desktop/trunk
+    rev = "512849";
+    sha256 = "1hl7znvv6qr4cwpkj8wlplpa63i1lhk2iax7hb4l1s1a4mijx9ls";
+  };
+
+  animatedStickersPatch = fetchpatch {
+    url = "https://github.com/kotatogram/kotatogram-desktop/commit/a196b0aba723de85ee573594d9ad420412b6391a.patch";
+    sha256 = "1mzfvcgj42r05k27423hbkvgg0gjrm3xv9hvadfqaq1ys46bpqi0";
+  };
+
+  patches = [
+    ./tdesktop.patch
+    "${archPatches}/no-gtk2.patch"
+    ./Use-system-wide-font.patch
+    animatedStickersPatch
+  ];
+
+  postPatch = ''
+    substituteInPlace Telegram/SourceFiles/platform/linux/linux_libs.cpp \
+      --replace '"gtk-3"' '"${gtk3}/lib/libgtk-3.so"'
+    substituteInPlace Telegram/SourceFiles/platform/linux/linux_libs.cpp \
+      --replace '"appindicator3"' '"${libappindicator-gtk3}/lib/libappindicator3.so"'
+    substituteInPlace Telegram/SourceFiles/platform/linux/linux_libnotify.cpp \
+      --replace '"notify"' '"${libnotify}/lib/libnotify.so"'
+  '';
+
+  # We want to run wrapProgram manually (with additional parameters)
+  dontWrapGApps = true;
+  dontWrapQtApps = true;
+
+  nativeBuildInputs = [ pkgconfig pythonPackages.gyp cmake ninja gcc9 ];
+
+  buildInputs = [
+    qtbase qtimageformats gtk3 libappindicator-gtk3 ffmpeg_4 openalSoft lzma
+    lz4 zlib minizip openssl libtgvoip rlottie-tdesktop range-v3
+  ];
+
+  enableParallelBuilding = true;
+
+  GYP_DEFINES = concatStringsSep "," [
+    "DESKTOP_APP_DISABLE_CRASH_REPORTS"
+    "TDESKTOP_DISABLE_AUTOUPDATE"
+    "TDESKTOP_DISABLE_REGISTER_CUSTOM_SCHEME"
+    "TDESKTOP_DISABLE_DESKTOP_FILE_GENERATION"
+  ];
+
+  NIX_CFLAGS_COMPILE = [
```

Comment:
Are you sure that this can't be done without setting `NIX_CFLAGS_COMPILE` manually?

Reply:
Yes. tdesktop build scripts don't support pkgconfig because they are optimized for static compilation.

Reply:
I'm still inclined to believe that those `-I` flags shouldn't be passed manually.

Reply:
I cleaned them as much as possible
Compare with https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix#L73

Reply:
Ah, interesting. I wonder why this is not handled by `addPackage` in `stdenv/setup`.

Reply:
Because there are no support for pkgconfig in tdesktop build scripts.